keras_rcnn/layers/losses.py

```diff
@@ -1,45 +1,38 @@
 import keras.layers
 
+import keras_rcnn.backend
 
-class Classification(keras.layers.Layer):
+
+class ClassificationLoss(keras.layers.Layer):
     def __init__(self, anchors, **kwargs):
         self.anchors = anchors
 
-        self.is_placeholder = True
-
-        super(Classification, self).__init__(**kwargs)
+        super(ClassificationLoss, self).__init__(**kwargs)
 
-    def _loss(self, y_true, y_pred):
-        # Binary classification loss
-        x, y = y_pred[:, :, :, :], y_true[:, :, :, self.anchors:]
+    def _loss(self, rpn_labels, rpn_classification):
+        rpn_classification = keras.backend.reshape(rpn_classification, [-1, 2])
+        rpn_classification = keras.backend.gather(rpn_classification, keras_rcnn.backend.where(keras.backend.not_equal(rpn_labels, -1)))
+        rpn_labels         = keras.backend.gather(rpn_labels, keras_rcnn.backend.where(keras.backend.not_equal(rpn_labels, -1)))
+        loss               = keras.backend.mean(keras.backend.categorical_crossentropy(rpn_labels, rpn_classification))
```

Comment:
This should be `sparse_categorical_crossentropy` not `categorical_crossentropy`. Since your inputs are of shape rpn_labels.shape = (None,) and rpn_classification.shape = (None, 2).